### ISSUE TITLE
Auto mark personal chats as read if read receipts are disabled

### ISSUE THREAD

**Issue by izzues** [tomer8007/whatsapp-web-incognito#16]
I've been using WhatsApp with read receipts turned off, and this extension has been very useful for staying invisible in group chats, viewing statuses and playing media. But for personal conversations it's actually become annoying, because to get rid of the unread count I need to either reply to the person or click "mark as read" and confirm it in the popup, rather than just viewing the conversation.

So it would be nice if the extension could detect whether read receipts are turned on or off, and if they're off it could just mark personal chats as read as you view them. I'm not sure if this automatic detection is possible, so just adding it as an option would already be very useful!

As always, thanks a lot for your work!

**Comment by tomer8007** [comment]
Sending read receipts by default to all personal chats even when the blocking is on would probably not be what most people expect, but I agree the popup could be annoying when you actually have an ongoing conversation.

I think instead we could add the option to never show the popup again, which connects to your previous issue.

**Comment by izzues** [comment]
Hello, tomer! Thanks again for your replies.

Maybe I wasn't clear enough. I don't know what most people expect, but isn't being "invisible" to others the only purpose of the extension? Other people don't see the read receipts you've sent if you have disabled them on the app, right? If the extension did what I propose, you would still be invisible, the only difference would be that _you_ would be able to view the conversations you visited as read, without having to mark them as such. So currently hiding the popup is not enough, you still have to mark the conversation as read, which would be unnecessary if the extension did what I described.

**Comment by tomer8007** [comment]
So if I understand you correctly, you just want to get rid of the green unread counter whenever you _really_ read a personal chat so that it won't confuse you later?

**Comment by izzues** [comment]
Yes, that's pretty much it :)

**Comment by tomer8007** [comment]
Okay, 
So to handle this problem I initially thought that since completely eliminating the unread counter in the web version does not impact your phone (which may or may not be confusing), changing its color from green to red ('read and blocked' mode) could be a fair enough indicator.

Now the problem with this is that maybe this still looks like it's unread. You see, from WhatsApp's standpoint, unread counter in the GUI means that read receipts were not sent and vice versa. 

What do you think?

**Comment by izzues** [comment]
Sorry for the delay!

My previous reply wasn't completely accurate. I not only want to "get rid" of the counter on the web version, but also on my phone, so I suppose the only solution (for me, but I understand if others don't like it) really is to send the receipts. The red color on the web version sure would help (though to be honest I'd prefer grey or any other dimmed color), but it still wouldn't be enough since I do pick up my phone from time to time and still get confused by notifications and messages on the widget and the app itself.

**Comment by tomer8007** [comment]
So basically what's the possible enhancement here? Sending read receipts when you read messages even when blocking is turned on is just lying.

**Comment by izzues** [comment]
As I said, at least for me, the utility of the extension has nothing to do with sending read receipts or not, but simply with _beeing seen_ by other users or not. So of course if you have read receipts turned on in the app you should also not send them via web. But if you have them disabled, no one is going to see them, whether they're sent or not. Having "blocking" turned on in this context is still useful as long as it keeps people from seeing me online (and as I said, the blocking is still useful for viewing groups, statuses and media).

So don't see any advantage in _not_ sending receipts if people are not going to notice any difference. Maybe other people have different concerns (e.g. they don't want even WhatsApp itself to know if you've seen the message, but that would be pretty pointless if you're still going to use the app).

In short, the enhancement the extension should provide (from my point of view) is _not being seen in situations that are not being covered by disabling read receipts in the app_. Hopefully I made things clearer!

**Comment by tomer8007** [comment]
> But if you have them disabled, no one is going to see them, whether they're sent or not

Umm still not sure I got you here. If I turn off blocking read receipts, whoever the read receipt was sent to will see it.

By saying "invisible" you mean blocking read receipts for groups (plus statuses), and blocking "Last Seen", right?  Your problem is that you **do** want read receipts to be sent in private conversations, but you don't want to become "visible", am I correct?

**Comment by izzues** [comment]
Alright, so here's how I see it:
* On WhatsApp's side, you can either enable or disable read receipts
* On the extension's, you can either enable or disable "invisible mode" (I'll call it that instead of "blocking")

My proposal is:
* When "invisible mode" is disabled in the extension, don't do anything (allow sending all receipts)
* When "invisible mode" is enabled in the extension and receipts are enabled on WhatsApp, block *all* receipts from being sent
* When "invisible mode" is enabled in the extension and receipts are disabled on WhatsApp, block all receipts from being sent, *except those in private conversations*

My point is that, whether you send receipts or not when "invisible mode" is enabled in the extension and receipts are disabled in WhatsApp, in terms of "invisibility" the end result is the same: other people won't see the receipts as read (... or am I wrong?), so sending the receipts in personal chats has no downsides and has the upside of letting you see them as read.

Anyway, it's just a tiny UX thing (and inconsistent, since you can't have it for group chats), so it's ok if you think it's not a good idea.

**Comment by tomer8007** [comment]
Oh, thank you for this eplanation, now I finally understand (didn't realise you are referring the the WhatsApp fetaure).

So yeah, this is a good idea for people that have read receipts disabled on their phone, and it only depends on the technical abillity to detect this from the Web. I think it's possible. Will take a look at it.

**Comment by izzues** [comment]
Awesome! Sorry for not being clear enough for so long and thank you so much for your attention!

**Comment by tomer8007** [comment]
So it looks like WhatsApp Web always tries to sent out the `read` request when you view a personal chat, regardless of the read receipts option on th phone. It is then the phone that doesn't relay the request in case read receipts are disabled.

Also I didn't see any packets coming in to the web when I enabled\disabled that option, and therefore I think such a feature is not really possible.

**Comment by izzues** [comment]
Oh, what a pity! Thanks for looking into this anyway! Feel free to close the issue :)
